ga4gh/datamodel/genotype_phenotype.py

```diff
@@ -0,0 +1,467 @@
+            else:
+                a[row['p']] = obj
+
+            a['id'] = row['s']
+        return a
+
+    def to_ga4gh(self, annotation):
+        """
+        given an annotation dict, return a protocol.FeaturePhenotypeAssociation
+        """
+
+        fpa = None
+
+        # annotation keys
+        # hasSubject = 'http://purl.org/oban/association_has_subject'
+        # hasPredicate = 'http://purl.obolibrary.org/obo/RO_has_environment'
+        source = 'http://purl.org/dc/elements/1.1/source'
+        location = 'location'
+        _evidence = 'http://purl.obolibrary.org/obo/RO_0002558'
+        hasObject = 'http://purl.org/oban/association_has_object'
+        # location keys
+        GENO_0000408 = 'http://purl.obolibrary.org/obo/GENO_0000408'
+        # GENO_reference_amino_acid = 'http://purl.obolibrary.org/obo/'
+        # 'GENO_reference_amino_acid'
+        # _type = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'
+        # _location = 'http://biohackathon.org/resource/faldo#location'
+        # GENO_results_in_amino_acid_change = 'http://purl.obolibrary.org/obo/'
+        # 'GENO_results_in_amino_acid_change'
+        # RO_0002200 = 'http://purl.obolibrary.org/obo/RO_0002200'
+        # RO_0002205 = 'http://purl.obolibrary.org/obo/RO_0002205'
+        # label = 'http://www.w3.org/2000/01/rdf-schema#label'
+
+        location = annotation['location']
+        if(GENO_0000408 in location):
+            (_id, _ontologySource) = self.namespace_split(
+                                        location[GENO_0000408]['val'])
+            name = location[GENO_0000408]['label']
+        else:
+            (_id, _ontologySource) = self.namespace_split(location['id'])
+            name = location['id']
+
+        d = {
+          "featureType": {
+              "name": name,
+              "id": _id,
+              "ontologySource": _ontologySource},
+          "parentIds": [],
+          "featureSetId": "",
+          "attributes": {"vals": {}},
+          "id": annotation["id"],
+        }
+        if (not protocol.Feature.validate(d)):
+            raise exceptions.RequestValidationFailureException(d,
+                                                               protocol.
+                                                               Feature)
+
+        f = protocol.Feature.fromJsonDict(d)
+
+        (_id, _ontologySource) = self.namespace_split(
+                                       annotation[hasObject]['val'])
+
+        d = {
+            "features": [],
+            "evidence": [],
+            "environmentalContexts": [],
+            "phenotype": {
+                "id": "",
+                "type": {"ontologySource": _ontologySource,
+                         "id": _id,
+                         "name": annotation[hasObject]['label']}
+                },
+            "id": annotation['id'],
+            "description": "null"
+        }
+
+        fpa = protocol.FeaturePhenotypeAssociation.fromJsonDict(d)
```

Comment:
I think this should be replace with something like:
```python
fpa = protocol.FeaturePhenotypeAssociation()
fpa.id = annotation['id']
fpa.features = []
fpa.description = None
# etc
```
I think if we get rid of the intermediate dictionaries and just update the GA object _directly_ this will simplify the code a lot.